go-controller/pkg/libovsdbops/chassis_test.go

```diff
@@ -0,0 +1,106 @@
+package libovsdbops
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/sbdb"
+	libovsdbtest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/libovsdb"
+)
+
+func TestDeleteChassis(t *testing.T) {
+	uuid := "b9998337-2498-4d1e-86e6-fc0417abb2f0"
+	tests := []struct {
+		desc             string
+		chassis          *sbdb.Chassis
+		chassisPredicate chassisPredicate
+		initialDB        []libovsdbtest.TestData
+		expectedDB       []libovsdbtest.TestData
+	}{
+		{
+			desc:    "delete chassis and chassis private",
+			chassis: &sbdb.Chassis{Name: "test"},
+			initialDB: []libovsdbtest.TestData{
+				&sbdb.Chassis{Name: "test"},
+				&sbdb.ChassisPrivate{Name: "test"},
+			},
+			expectedDB: []libovsdbtest.TestData{},
+		},
+		{
+			desc:    "delete chassis and chassis private by UUID",
+			chassis: &sbdb.Chassis{UUID: uuid},
+			initialDB: []libovsdbtest.TestData{
+				&sbdb.Chassis{UUID: uuid, Name: "test"},
+				&sbdb.ChassisPrivate{Name: "test"},
+			},
+			expectedDB: []libovsdbtest.TestData{},
+		},
+		{
+			desc:    "delete chassis when chassis private does not exist",
+			chassis: &sbdb.Chassis{Name: "test"},
+			initialDB: []libovsdbtest.TestData{
+				&sbdb.Chassis{Name: "test"},
+			},
+			expectedDB: []libovsdbtest.TestData{},
+		},
+		{
+			desc:    "delete chassis private when chassis does not exist",
+			chassis: &sbdb.Chassis{Name: "test"},
+			initialDB: []libovsdbtest.TestData{
+				&sbdb.ChassisPrivate{Name: "test"},
+			},
+			expectedDB: []libovsdbtest.TestData{},
+		},
+		{
+			desc:             "delete chassis and chassis private by predicate",
+			chassisPredicate: func(c *sbdb.Chassis) bool { return c.Hostname == "testNode" },
+			initialDB: []libovsdbtest.TestData{
```

Comment:
nice!

Reply:
NIT: can we add a TC where one of the 2 rows get deleted, so we make sure the predicate is applied properly?

Reply:
```
		{
			desc:             "delete only one chassis and chassis private by predicate",
			chassisPredicate: func(c *sbdb.Chassis) bool { return c.Hostname == "testNode" },
			initialDB: []libovsdbtest.TestData{
				&sbdb.Chassis{Name: "test", Hostname: "testNode"},
				&sbdb.Chassis{Name: "test2", Hostname: "testNode2"},
				&sbdb.ChassisPrivate{Name: "test"},
				&sbdb.ChassisPrivate{Name: "test2"},
			},
			expectedDB: []libovsdbtest.TestData{
				&sbdb.Chassis{Name: "test2", Hostname: "testNode2"},
				&sbdb.ChassisPrivate{Name: "test2"},
			},
		},
```